yt_dlp/utils.py

```diff
@@ -5426,18 +5428,20 @@ def apply_path(start_obj, path):
 
         return has_branched, objs
 
-    def _traverse_obj(obj, path):
+    def _traverse_obj(obj, path, last=True):
         has_branched, results = apply_path(obj, path)
         results = LazyList(x for x in map(type_test, results) if x is not None)
-        if results:
-            return results.exhaust() if get_all and has_branched else results[0]
+        if not results:
+            return [] if has_branched and last and default is NO_DEFAULT else None
 
-    for path in paths:
-        result = _traverse_obj(obj, path)
+        return results.exhaust() if get_all and has_branched else results[0]
```

Comment:
This is clearer imo:
```py
        if get_all and has_branched:
            if results or (last and default is NO_DEFAULT):
                return results.exhaust()
        elif results:
            return results[0]
```

Reply:
this can be further simplified if you set `last = index == len(paths) and default is NO_DEFAULT`, but ur call